src/Portage/EBuild.hs

```diff
@@ -118,17 +102,15 @@ showEBuild now ebuild =
   ss ("# ebuild generated by hackport " ++ hackportVersion ebuild). nl.
   sconcat (map (\(k, v) -> ss "#hackport: " . ss k . ss ": " . ss v . nl) $ used_options ebuild).
   nl.
+  (case cabal_pn ebuild of
+     Nothing -> id
+     Just pn -> ss "CABAL_PN=". quote pn. nl.).
```

Comment:
Same here, and then remove the second function composition operator `(.)`. That should hopefully work.

Reply:
I think I got it figured out. That module is admittedly pretty confusing syntactically.